cdap-cli/src/main/java/co/cask/cdap/cli/command/security/CheckActionCommand.java

```diff
@@ -46,29 +48,28 @@ public CheckActionCommand(AuthorizationClient client, CLIConfig cliConfig) {
   @Override
   public void perform(Arguments arguments, PrintStream output) throws Exception {
     EntityId entity = EntityId.fromString(arguments.get(ArgumentName.ENTITY.toString()));
-    String user = arguments.get("user");
-    Set<Action> actions = fromStrings(Splitter.on(",").split(arguments.get("actions")));
+    String principalName = arguments.get("principal-name");
+    Principal.PrincipalType principalType = Principal.PrincipalType.valueOf(arguments.get("principal-type"));
+    Principal principal = new Principal(principalName, principalType);
+    Action action = Action.valueOf(arguments.get("action"));
 
-    boolean authorized = client.authorized(entity, user, actions);
-    output.printf("%s\n", authorized);
+    try {
+      client.authorized(entity, principal, action);
+    } catch (UnauthorizedException e) {
+      output.printf("Principal %s is not authorized to perform action %s on entity %s\n", principal, action, entity);
```

Comment:
uh oh.
put a return statement, else it'll print both statements (line 59 and line 61).